### PR TITLE
Added support for negative jobs values

### PR DESCRIPTION
**Pull request by OLUWAMUYIWA**
Ref: #495

### CODE REVIEW

File: cargo-nextest/src/dispatch.rs [hunk 1]
```diff
@@ -1579,4 +1580,21 @@ mod tests {
...
+
+        let pos_j = "cargo nextest run --jobs 3";
+        let _app =
+            CargoNextestApp::try_parse_from(shell_words::split(pos_j).expect("valid command line"))
+                .unwrap();
+
+        let zero_j = "cargo nextest run --jobs 0";
+
+        assert!(CargoNextestApp::try_parse_from(shell_words::split(zero_j).unwrap()).is_err());
+    }
```
Comment (sunshowers) [inline cargo-nextest/src/dispatch.rs:1599]: These test vectors can just go in `test_argument_parsing` -- no need for another function.

File: cargo-nextest/src/dispatch.rs [hunk 2]
```diff
@@ -594,6 +594,7 @@ impl CargoOptions {
 /// Test runner options.
 #[derive(Debug, Default, Args)]
 #[clap(next_help_heading = "RUNNER OPTIONS")]
+#[clap(allow_negative_numbers = true)]
```
Comment (sunshowers) [inline cargo-nextest/src/dispatch.rs:597]: Could you move this to just the specific argument where negative numbers are allowed? i.e. the `test_threads` argument.

**Review by sunshowers — COMMENTED** [review]
This looks absolutely fantastic, thank you! Just a couple of really, really minor comments.

**Comment by OLUWAMUYIWA** [comment]
Fixed, @sunshowers. I responded late because we had a power outage lasting days. Hope this resolves your comments.

File: cargo-nextest/src/dispatch.rs [hunk 3]
```diff
@@ -606,7 +606,8 @@ pub struct TestRunnerOpts {
         visible_alias = "jobs",
         value_name = "THREADS",
         conflicts_with_all = &["no-capture", "no-run"],
-        env = "NEXTEST_TEST_THREADS"
+        env = "NEXTEST_TEST_THREADS",
+	allow_hyphen_values = true
```
Comment (sunshowers) [inline cargo-nextest/src/dispatch.rs:610]: Oh I see what happened here -- `allow_negative_numbers` isn't supported on `Arg` in clap 3 (but is in clap 4 -- https://epage.github.io/blog/2022/09/clap4/#fixing-parsing-for-hyphenated-values). I'll land this and then revert the change.

File: site/help-text/run-help.txt [hunk 4]
```diff
@@ -18,8 +18,10 @@ OPTIONS:
 
 RUNNER OPTIONS:
         --no-run                    Compile, but don't run tests
-    -j, --test-threads <THREADS>    Number of tests to run simultaneously [possible values: integer
-                                    or "num-cpus"] [env: NEXTEST_TEST_THREADS=] [aliases: jobs]
+    -j, --test-threads <THREADS>    Number of tests to run simultaneously [possible values: integer or "num-cpus".
```
Comment (jiangying000) [inline site/help-text/run-help.txt:21]: this change is not modified in dispatch.rs, so ./scripts/regenerate-help-text.sh reverted the change, I going to fix in #518 

@sunshowers

**Pull request merged**